src/Symfony/Component/Workflow/Registry.php

```diff
@@ -22,26 +23,32 @@ class Registry
     private $workflows = array();
 
     /**
-     * @param Workflow $workflow
-     * @param string   $className
+     * @param Workflow                        $workflow
+     * @param string|SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $className)
+    public function add(Workflow $workflow, $supportStrategy)
     {
-        $this->workflows[] = array($workflow, $className);
+        if (!$supportStrategy instanceof SupportStrategyInterface) {
+            @trigger_error('Support of class name string was deprecated after version 3.2 and won\'t work anymore in 4.0.', E_USER_DEPRECATED);
+        }
+
+        $this->workflows[] = array($workflow, $supportStrategy);
     }
 
     /**
      * @param object      $subject
      * @param string|null $workflowName
      *
      * @return Workflow
+     *
+     * @throws InvalidArgumentException
```

Comment:
we don't add this phpdoc when we cannot document in which case the exception is thrown